src/utils/message.ts

```diff
@@ -59,42 +64,41 @@ const calculateLrc = ( data: Uint8Array, initialLrc = 0x55 ) => {
     return lrc;
 };
 
-const getCommand = ( id: number, data: Uint8Array, direction = TYPE_AUTO ): Command => {
-    // given direction
-    if ( direction === TYPE_DOWNLINK || direction === TYPE_UPLINK ) {
-        const commandsById = direction === TYPE_DOWNLINK ? downlinkCommandsById : uplinkCommandsById;
-        const command = commandsById[id];
+const getCommand = ( id: number, data: Uint8Array, direction = DIRECTION_TYPE_AUTO ): Command => {
+    if ( !directionTypeIds.has(direction) ) {
+        throw new Error('wrong direction type');
+    }
 
-        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-        if ( !command ) {
-            throw new Error(`Unsupported command with id: ${id}.`);
-        }
+    const downlinkCommand = downlinkCommandsById[id];
+    const uplinkCommand = uplinkCommandsById[id];
+
+    // check command availability
+    if (
```

Comment:
```suggestion
    if ( !downlinkCommand && !uplinkCommand ) { return new UnknownCommand({id, data}); }
```

else if ( direction === DIRECTION_TYPE_DOWNLINK && !downlinkCommand ) return UnknownDownlinkCommand or throw an Exception immediately

Reply:
we don't need to separate up/down UnknownCommand